src/sdam/topology.ts

```diff
@@ -556,22 +556,20 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
     const waitQueueMember: ServerSelectionRequest = {
       serverSelector,
       transaction,
-      callback
+      callback,
+      timeoutController: new TimeoutController(options.serverSelectionTimeoutMS)
     };
 
-    const serverSelectionTimeoutMS = options.serverSelectionTimeoutMS;
-    if (serverSelectionTimeoutMS) {
-      waitQueueMember.timer = setTimeout(() => {
-        waitQueueMember[kCancelled] = true;
-        waitQueueMember.timer = undefined;
-        const timeoutError = new MongoServerSelectionError(
-          `Server selection timed out after ${serverSelectionTimeoutMS} ms`,
-          this.description
-        );
+    waitQueueMember.timeoutController.signal.onabort = () => {
+      waitQueueMember[kCancelled] = true;
```

Comment:
this handler is passed an DOM Event object, do we need to call preventDefault? (I have no idea, need to study up on web APIs)

Reply:
No, prevent default stops the event handling chain.  In browser APIs, it's most commonly used to prevent the default behavior when a DOM element is clicked / handled (the `preventDefault` docs, for example, demonstrate how it can be used to enable a click handler from preventing a checkbox from being "checked" when it is clicked on)